### ISSUE TITLE
Static default site name in installer

### ISSUE THREAD

**Issue by mikemccaffrey** [backdrop/backdrop-issues#2291]
As @quicksketch just pointed out in his demo, it doesn't make a lot of sense to use the domain name as the default site name in the installer. If you are using pantheon for hosting, it will use something like `dev-new-sitename.pantheonsite.io` as the name, and if you are using local vhosts you might get something like `~^(www\.)?(?<sname>.+?).localhost.dev$`.

Should we perhaps just use a static default value like "My Site" or "My Backdrop Site"?

**Comment by klonos** [comment]
Related: #2209

In that issue, one idea was to have the site name set to a default ("Backdrop CMS" for example), until the installer reaches the point where the user can specify a custom site name. Once that's done, the default, placeholder site name will be overwritten by whatever was set as the actual site name. From then on, the new, custom site name set by the user will be appended to the `<title>` tag.

One really bad thing with having the site name be something static, is that the browser tab text will also be static and literally the same on every tab if you have multiple installations going on at the same time. So, you won't be able to tell which tab is which installation unless you switch from tab to tab in order to see the URL. I think that having a unique identifier appended to the `<title>` tag (such as the site URL) makes a lot of sense from that point of view. Once the site name is set to something unique by the user, that will be used instead.

**Comment by klonos** [comment]
...which begs this question with regards to DX: if the site name is the same between the dev and live sites, how does one tell apart their tabs in a browser (without having to actually switch to each one of them)? Until now, I have never used such a setup (dev/live), but if I did, then I would be frustrated by the bad UX of having no way to tell which tabs are my local dev and which are the live. Haven't you guys ever run into this situation? What could make things better in your opinion? Perhaps have a setting ("This is a local development instance" checkbox) under Configuration > Development or under Configuration > System > Site info that adds an overlaid icon to the favicon or something?

**Comment by olafgrabienski** [comment]
@klonos Re UX dev/live: In Drupal, I use https://www.drupal.org/project/environment_indicator:

> The Environment Indicator adds a coloured bar on the site informing you which environment you're currently in (Development, Staging, Production, etc.).

It's possible to use the admin menu as the "coloured bar". Find it really helpful. There is also an option to generate and show environment specific favicons.

In Backdrop, I don't know a similar tool so far. In the frontend, if you have a subtheme, you could add some CSS to the dev theme to show a DEV button at the top of the page. And to recognize the different sites in the browser tabs, you could change the Site name on dev (My site vs. DEV My site).

**Comment by klonos** [comment]
@olafgrabienski a solution like that (although very useful) would still require the person to switch between browser tabs in order to see the indicator. The point is be to add some indication on the actual browser tab itself. Something that would either change the tab text or the favicon perhaps.

**Comment by olafgrabienski** [comment]
@klonos I see! As I pointed out, Environment Indicator **does** provide the option to show specific favicons.

**Comment by klonos** [comment]
> As I pointed out, Environment Indicator **does** provide the option to show specific favicons.

Yes, sorry @olafgrabienski missed that somehow. It seems that this is a really well-thought solution then. Thanx,

**Comment by klonos** [comment]
@mikemccaffrey, @olafgrabienski I have filed a PR in #2209 that might be addressing some of your concerns expressed here. Would appreciate the feedback when you get a chance.

**Comment by olafgrabienski** [comment]
@klonos Thanks for the update. Had no concerns, actually.

---

... Pointed only to the Drupal Environment Indicator module which solves the following issue: 

> Until now, I have never used such a setup (dev/live), but if I did, then I would be frustrated by the bad UX of having no way to tell which tabs are my local dev and which are the live.

Btw, in the meantime a similar Backdrop module has been released: [Simple Environment Indicator](https://backdropcms.org/project/simplei)

> This module helps developers see at a glance which Backdrop CMS site environment they are on. It does this by colouring the adminbar; putting the environment name in the adminbar; and putting an indicator in the favicon.
